### PR TITLE
Add stories related to auth flow

### PR DESCRIPTION
**Pull request by matjack1**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `master` branch of Chapter.

Since in the DB we have user email/password and social providers we need to add the user stories for the auth flow.

I'm proposing the smallest set of features I can think of to the auth flow to the MVP. I'm proposing to only use Github as a social provider to minimise things and also I think for now we can only target developers (or aspiring ones!) working on the MVP :)

### CODE REVIEW

**Comment by chrismgonzalez** [comment]
@matjack1 Thanks for your work on this PR.  Have you thought about potential non-tech users of the application that may not have a Github?  I would like to know the thoughts of others with regards to at least having Facebook or Google logins available in the MVP in addition to what you are proposing with Github integration.   

It seems to be pretty quiet around here as of this evening so we'll see if anyone else chimes in.

**Comment by iansltx** [comment]
+1 for FB/Google login capabilities. Between FB/Google/GitHub there's generally going to be one option that folks use and don't mind linking to another account (my personal go-to is Google, with GitHub only getting used on more tech-y stuff).

File: README.md [hunk 1]
```diff
@@ -46,7 +46,17 @@ We will focus on building an open API first. Then developers can use the API to
...
 ## User stories so far
 
-Our goal is to keep things simple and not reinvent wheels. So far we have only two user roles: participants and chapter organizers.
+Our goal is to keep things simple and not reinvent wheels. So far we have only two user roles: participants and chapter organizers, both of which are users.
+
+### As a user
+
+- I can open a registration page where I can sign up with email password.
+
+- I can log in with my email/password or I can log in with social login via Github.
```
Comment (chrismgonzalez) [inline README.md]: User can login via email/password, Github, Facebook, or Google account

**Comment by matjack1** [comment]
OK, I've updated the PR to include Google and FB. I thought starting only with Github to keep things simple. We can always add more later.

For the MVP I would have started with the email only if it was for me, much less code to start validating everything else, which I think is more important. But if you all agree with the three social logins I'm fine.

**Comment by vkWeb** [comment]
@matjack1 @chrismgonzalez Let's just start with google to keep things simple. What do you say?

**Comment by chrismgonzalez** [comment]
Sure that sounds good!  Thanks guys.

**Comment by matjack1** [comment]
@chrismgonzalez @vkWeb makes sense, PR updated and issue too :)

File: README.md [hunk 2]
```diff
@@ -46,7 +46,17 @@ We will focus on building an open API first. Then developers can use the API to
...
-Our goal is to keep things simple and not reinvent wheels. So far we have only two user roles: participants and chapter organizers.
+Our goal is to keep things simple and not reinvent wheels. So far we have only two user roles: participants and chapter organizers, both of which are users.
+
+### As a user
+
+- I can open a registration page where I can sign up with email password.
+
+- I can log in with my email/password or I can log in with social login via Google.
+
+- I can see my account page where I can reset my password if I've registered with email/password. Otherwise I'll see a link to my Google profile.
```
Comment (vkWeb) [inline README.md]: @matjack1 `email/password` sounds like email or password. Please replace `email/password` with `email and password` to keeps things clear.

Btw thanks for these changes :)

File: README.md [hunk 3]
```diff
@@ -46,7 +46,17 @@ We will focus on building an open API first. Then developers can use the API to
 
 ## User stories so far
 
-Our goal is to keep things simple and not reinvent wheels. So far we have only two user roles: participants and chapter organizers.
+Our goal is to keep things simple and not reinvent wheels. So far we have only two user roles: participants and chapter organizers, both of which are users.
+
+### As a user
+
+- I can open a registration page where I can sign up with email password.
```
Comment (wuilliam321) [inline README.md]: email and password. Details about min requirements?

File: README.md [hunk 4]
```diff
@@ -46,7 +46,17 @@ We will focus on building an open API first. Then developers can use the API to
...
+
+- I can log in with my email/password or I can log in with social login via Google.
```
Comment (wuilliam321) [inline README.md]: We can discuss if we can enable multiple options other than Google

File: README.md (hunk 2 above)
Comment (matjack1) [reply README.md]: changed, you're welcome :)

File: README.md (hunk 1 above)
Comment (matjack1) [reply README.md]: we decided to use Google only for now and once that is implemented and working we can add the others

File: README.md (hunk 3 above)
Comment (matjack1) [reply README.md]: I don't think they are necessary for now, we can start implementing something very minimal and then build on top of it. So I would not add no requirements now.

File: README.md (hunk 4 above)
Comment (matjack1) [reply README.md]: We have discussed above and decided to start with Google for now and once that is implemented we can add the others.

**Comment by matjack1** [comment]
I've updated this PR following everyone's feedback.

Summary is that I'm proposing to add email and password login without any further spec for now. We can add complexity requirements at a later stage (https://github.com/freeCodeCamp/chapter/issues/84#issuecomment-544151824).

Regarding social login I'm proposing to start with Google only and then we add the others at a later stage (https://github.com/freeCodeCamp/chapter/issues/84#issuecomment-544151824).

**Comment by vkWeb** [comment]
@chrismgonzalez This is ready. Please squash merge :)

**Pull request merged**